de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -1014,12 +1018,17 @@ public void extendContract(final @NonNull ContractExtendingRequest context)
 	{
 		Services.get(ITrxManager.class).run(ITrx.TRXNAME_ThreadInherited, localTrxName -> {
 
+			final Map<Integer, String> seenFlatrateCondition = new HashMap<>();
```

Comment:
use LinkedHashMap to preserve the same order.
Preserving the order might be useful for user to understand how the loop was created.